nav-app/src/app/services/data.service.ts

```diff
@@ -220,6 +229,19 @@ export class DataService {
                 }
             }
         }
+        //create matrices, which also creates tactics and filters techniques
+        for (let i = 0; i < matricesList.length; i++) {
+            let techniquesList = [];
+            if (matricesList[i].x_mitre_deprecated) {
+                continue;
+            }
+            for (let j = 0; j < domain.techniques.length; j++) {
+                if(domain.techniques[j].get_technique_domain(techniqueSDOslist[j]) == matricesList[i].external_references[0].external_id) {
+                    techniquesList.push(domain.techniques[j]);
+                }
```

Comment:
This logic should be updated to check if the matrix ID matches any of the domains listed in the technique's `x_mitre_domains` field. See comment above regarding only checking the first domain in the list of `x_mitre_domains`.